mlir/test/Conversion/ConvertToSPIRV/scf.mlir

```diff
@@ -45,3 +56,30 @@ func.func @while(%arg0: i32, %arg1: i32) -> i32 {
   }
   return %0 : i32
 }
+
+// CHECK-LABEL: @for
+// CHECK:       spirv.mlir.loop {
+// CHECK:         spirv.Branch ^[[HEADER:.*]](%{{.*}}, %{{.*}}, %{{.*}} : i32, f32, f32)
+// CHECK:       ^[[HEADER]]
+// CHECK:         spirv.BranchConditional %{{.*}}, ^[[BODY:.*]], ^[[MERGE:.*]]
+// CHECK:       ^[[BODY]]
+// CHECK-DAG:     spirv.Store "Function"
+// CHECK-DAG:     spirv.Store "Function"
+// CHECK:       spirv.Branch ^[[HEADER]]
+// CHECK:       ^[[MERGE]]
+// CHECK:         spirv.mlir.merge
+// CHECK:      }
+// CHECK-DAG:  spirv.Load "Function"
+// CHECK-DAG:  spirv.Load "Function"
```

Comment:
Same here

Reply:
This is similar to the `scf.while` case, except that the `scf.yield` is producing two results, which corresponds to two `spirv.Store "Function"` and `spirv.Load "Function"` pairs.